src/descriptor/mod.rs

```diff
@@ -68,6 +75,258 @@ pub enum Descriptor<Pk: MiniscriptKey> {
+                    is_wildcard = true;
+                    None
+                } else if is_wildcard {
+                    Some(Err(DescriptorKeyParseError(
+                        "'*' may only appear as last element in a derivation path.",
+                    )))
+                } else {
+                    Some(bip32::ChildNumber::from_str(p).map_err(|_| {
+                        DescriptorKeyParseError("Error while parsing key derivation path")
+                    }))
+                }
+            })
+            .collect::<Result<bip32::DerivationPath, _>>()?;
+
+        if (&derivation_path).into_iter().all(|c| c.is_normal()) {
+            Ok((xpub, derivation_path, is_wildcard))
+        } else {
+            Err(DescriptorKeyParseError(
+                "Hardened derivation is currently not supported.",
+            ))
+        }
+    }
+
+    /// Derives a new key using the path if self is a wildcard xpub. Otehrwise returns a copy of
+    /// self.
+    ///
+    /// Panics if derivation path contains a hardened child number
+    pub fn derive(&self, path: &[bip32::ChildNumber]) -> DescriptorPublicKey {
+        debug_assert!(path.into_iter().all(|c| c.is_normal()));
+
+        match *self {
+            DescriptorPublicKey::ShortPub(ref pk) => DescriptorPublicKey::ShortPub(pk.clone()),
+            DescriptorPublicKey::XPub(ref xpub) => {
+                if xpub.is_wildcard {
+                    DescriptorPublicKey::XPub(DescriptorXPub {
+                        origin: xpub.origin.clone(),
+                        xpub: xpub.xpub.clone(),
+                        derivation_path: (&xpub.derivation_path)
+                            .into_iter()
+                            .chain(path.iter())
+                            .cloned()
+                            .collect(),
+                        is_wildcard: false,
+                    })
+                } else {
+                    self.clone()
+                }
+            }
+        }
+    }
+}
+
+impl MiniscriptKey for DescriptorPublicKey {
+    type Hash = hash160::Hash;
+
+    fn to_pubkeyhash(&self) -> Self::Hash {
+        match *self {
+            DescriptorPublicKey::ShortPub(ref spub) => spub.key.to_pubkeyhash(),
+            DescriptorPublicKey::XPub(ref xpub) => {
+                let ctx = secp256k1::Secp256k1::verification_only();
+                xpub.xpub
+                    .derive_pub(&ctx, &xpub.derivation_path)
+                    .expect("Won't fail, only normal derivations")
+                    .public_key
+                    .to_pubkeyhash()
+            }
+        }
+    }
+}
+
+impl ToPublicKey for DescriptorPublicKey {
+    fn to_public_key(&self) -> bitcoin::PublicKey {
+        match *self {
+            DescriptorPublicKey::ShortPub(ref spub) => spub.key.to_public_key(),
+            DescriptorPublicKey::XPub(ref xpub) => {
+                let ctx = secp256k1::Secp256k1::verification_only();
```

Comment:
I think we can use the [global context](https://github.com/rust-bitcoin/rust-secp256k1/pull/224) by now.

Reply:
Hmm or maybe we need to have the context as part of the struct ?..

Reply:
We are working with public data, there is no reason not to use the global context imo.

Reply:
Ok, but it then makes me introduce a new `secp256k1` dependency separated from the `rust-bitcoin` one

Reply:
As long as it's the same as in the current rust-bitcoin version that seems ok, doesn't it? Cargo will just enable the feature, not add a whole new version of the crate.

Reply:
Ok so rust-secp256k1 v0.18 finally does not expose global-context https://github.com/rust-bitcoin/rust-secp256k1/blob/0.18.0/Cargo.toml, and v0.19 is a MSRV bump (and a rust-bitcoin bump as well).